Marlin/src/module/tool_change.cpp

```diff
@@ -467,7 +454,9 @@ inline void invalid_extruder_error(const uint8_t e) {
  * previous tool out of the way and the new tool into place.
  */
 void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool no_move/*=false*/) {
-  planner.synchronize();
+  #if DISABLED(MIXING_EXTRUDER)
+    planner.synchronize();
```

Comment:
Wouldn't we still want the synchronize to occur for mixing extruders? By removing the synchronize, the new mix might be applied to segments that have been queued, but not yet executed.

Reply:
No. The used mix for the blocks are copied into the block when it's populated.  Not just the index - but the complete information.

Reply:
Sounds fair